### ISSUE TITLE
Step width 8.33 kHz treated differently in regular scanning and spectrum analyzer

### ISSUE THREAD

**Issue by catequalsgood** [egzumer/uv-k5-firmware-custom#589]
This might be intentional but I noticed that in regular scanning mode a step width of 8.33 kHz always leads to 25 kHz increments after three steps, as expected. But in the spectrum analyzer the step seems to literally be treated as 8.330 kHz. This causes a growing mismatch between scanned frequency and valid air band channel frequencies. 
The attached image shows the result of entering the spectrum analyzer with a scanning range of 118.000 to 137.000 MHz showing invalid air band frequencies for both the current and highest frequency.

![IMG_1942](https://github.com/egzumer/uv-k5-firmware-custom/assets/70518425/e788b239-0286-4d51-9981-3cd39f0b32f5)

**Comment by prokrypt** [comment]
Definitely not supposed to happen. Someone forgot to `FREQUENCY_RoundToStep()` in a few places.

I had a go at fixing it @ https://github.com/prokrypt/uv-k5-firmware-custom/commit/0b6269480ebf90c47b7b14acf95c0f3159399508

**Comment by catequalsgood** [comment]
The upper limit is now displayed as "136.99999" but I can live with that. Other than that it seems to be working perfectly. Thanks!

**Comment by prokrypt** [comment]
> The upper limit is now displayed as "136.99999" but I can live with that. Other than that it seems to be working perfectly. Thanks!

Indeed. I figured it wasn't worth using extra bytes to fix such a small display quirk. Also, on my fork of the firmware the "last frequency" (137MHz) is not scanned anyways.
